frontend/lib/norent/components/subscribe.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useState } from "react";
+import { Trans, t } from "@lingui/macro";
+import { li18n } from "../../i18n-lingui";
+import { StaticImage } from "../../ui/static-image";
+import { getImageSrc } from "../homepage";
+
+const Subscribe = () => {
+  const [email, setEmail] = useState("");
+  const [isSuccessful, setSuccess] = useState(false);
+  const [response, setResponse] = useState("");
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setEmail(e.target.value);
+  };
+
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    const locale = li18n.language;
+
+    if (!email) {
+      setResponse(li18n._(t`Please enter an email address!`));
+      return;
+    }
+
+    fetch("/mailchimp/subscribe", {
+      method: "POST",
+      body: `email=${encodeURIComponent(
+        email
+      )}&language=${locale}&source=norent`,
+      headers: {
+        "Content-Type": "application/x-www-form-urlencoded",
+      },
+    })
+      .then((result) => result.json())
+      .then((result) => {
+        if (result.status === 200) {
+          setSuccess(true);
+          setResponse(li18n._(t`All set! Thanks for subscribing!`));
+        } else if (result.errorCode === "INVALID_EMAIL") {
+          setResponse(li18n._(t`Oops! That email is invalid.`));
+        } else {
+          window &&
+            window.Rollbar &&
+            window.Rollbar.error(
+              `Mailchimp email signup responded with error code ${result.errorCode}.`
+            );
```

Comment:
I guess this might be easier to read as just `window.Rollbar?.error()`?  I don't think you need to check for the existence of `window` because it will definitely exist if this `handleSubmit()` function it's in is ever called.

Reply:
cool cool